File: benchmark/c/opl/opl_emu_clamp/opl_emu_clamp.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <string.h>

int32_t opl_emu_clamp(int32_t value, int32_t minval, int32_t maxval);

int32_t opl_emu_clamp(int32_t value, int32_t minval, int32_t maxval)
{
	if (value < minval)
		return minval;
	if (value > maxval)
		return maxval;
	return value;
}

File: translation_result/real_world_code/success/c/opl/opl_emu_clamp/mistral/BaseRepair/opl_emu_clamp.rs

use std::i32;

fn opl_emu_clamp(value: i32, minval: i32, maxval: i32) -> i32 {
    if value < minval {
        return minval;
    } else if value > maxval {
        return maxval;
    }
    value
}
